plume-front/src/editor.rs

```diff
@@ -0,0 +1,244 @@
+use std::rc::Rc;
+use stdweb::{unstable::{TryInto, TryFrom}, web::{*, html_element::*, event::*}};
+use CATALOG;
+
+macro_rules! mv {
+    ( $( $var:ident ),* => $exp:expr ) => {
+        {
+            $( let $var = $var.clone(); )*
+            $exp
+        }
+    }
+}
+
+fn get_elt_value(id: &'static str) -> String {
+    let elt = document().get_element_by_id(id).unwrap();
+    let inp: Result<InputElement, _> = elt.clone().try_into();
+    let textarea: Result<TextAreaElement, _> = elt.try_into();
+    inp.map(|i| i.raw_value()).unwrap_or_else(|_| textarea.unwrap().value())
+}
+
+fn set_value<S: AsRef<str>>(id: &'static str, val: S) {
+    let elt = document().get_element_by_id(id).unwrap();
+    let inp: Result<InputElement, _> = elt.clone().try_into();
+    let textarea: Result<TextAreaElement, _> = elt.try_into();
+    inp.map(|i| i.set_raw_value(val.as_ref()))
+        .unwrap_or_else(|_| textarea.unwrap().set_value(val.as_ref()))
+}
+
+fn no_return(evt: KeyDownEvent) {
+    if evt.key() == "Enter" {
+        evt.prevent_default();
+    }
+}
+
+pub fn init() {
+    document().get_element_by_id("plume-editor")
+        .and_then(|ed| {
```

Comment:
this closure is really long. It would gain in readability to be divided in multiple smaller functions called one after an other (like "hide non-js form", "add js editor", "show popup", "copy editor to form"...). I understand what it's doing, but if there is something wrong, it's too complex to see it right away